src/main/java/org/spongepowered/mod/mixin/world/MixinWorld.java

```diff
@@ -260,4 +328,94 @@ public boolean doesKeepSpawnLoaded() {
     public void setKeepSpawnLoaded(boolean keepLoaded) {
         this.keepSpawnLoaded = keepLoaded;
     }
+
+    @Override
+    public void playSound(SoundType sound, Vector3d position, double volume) {
+        this.playSound(sound, position, volume, 1);
+    }
+
+    @Override
+    public void playSound(SoundType sound, Vector3d position, double volume, double pitch) {
+        this.playSound(sound, position, volume, pitch, 0);
+    }
+
+    @Override
+    public void playSound(SoundType sound, Vector3d position, double volume, double pitch, double minVolume) {
+        this.playSoundEffect(position.getX(), position.getY(), position.getZ(), sound.getName(), (float) Math.max(minVolume, volume), (float) pitch);
+    }
+
+    @Override
+    public Optional<Entity> getEntityFromUUID(UUID uuid) {
+        if ((Object) this instanceof WorldServer) {
+            // TODO Should this be done in an override in a WorldServer mixin?
+            return Optional.fromNullable((Entity) ((WorldServer) (Object) this).getEntityFromUuid(uuid));
+        }
+        for (net.minecraft.entity.Entity entity : this.loadedEntityList) {
+            if (entity.getUniqueID().equals(uuid)) {
+                return Optional.of((Entity) entity);
+            }
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public Optional<String> getGameRule(String gameRule) {
+        if (this.worldInfo.getGameRulesInstance().hasRule(gameRule)) {
+            return Optional.of(this.worldInfo.getGameRulesInstance().getGameRuleStringValue(gameRule));
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public void setGameRule(String gameRule, String value) {
+        this.worldInfo.getGameRulesInstance().setOrCreateGameRule(gameRule, value);
+    }
+
+    @Override
+    public Map<String, String> getGameRules() {
+        GameRules gameRules = this.worldInfo.getGameRulesInstance();
+        // TODO Could expose GameRules#theGameRules via reflection to remove the
+        // need to iterate over keys
+        Map<String, String> ruleMap = new HashMap<String, String>();
+        for (String rule : gameRules.getRules()) {
+            ruleMap.put(rule, gameRules.getGameRuleStringValue(rule));
+        }
+        return ruleMap;
+    }
+
+    @Override
+    public long getWorldSeed() {
+        return this.getSeed();
+    }
+
+    @Override
+    public void setSeed(long seed) {
+        ObfuscationReflectionHelper.setPrivateValue(WorldInfo.class, this.worldInfo, seed, "randomSeed");
```

Comment:
@bloodmc do we want to use AT's for things like this to prevent errors during MC version updates?